src/IECoreDelight/Renderer.cpp

```diff
@@ -1292,7 +1480,28 @@ class DelightRenderer final : public IECoreScenePreview::Renderer
 
 		ObjectInterfacePtr light( const std::string &name, const IECore::Object *object, const AttributesInterface *attributes ) override
 		{
-			return this->object( name, object, attributes );
+			if( object )
+			{
+				return this->object( name, object, attributes );
+			}
+
+			const IECore::MessageHandler::Scope s( m_messageHandler.get() );
+
+			auto a = static_cast<const DelightAttributes *>( attributes );
+			if( const ShaderNetwork * network = a->usdLightShader() )
+			{
+				if( const Shader *outShader = network->outputShader() )
+				{
+					if( outShader->getName() == "SphereLight" )
```

Comment:
All these conditionals mean that if a location doesn't have the required attributes when first seen by IPR, it will never become a light. For instance, if you start IPR with an Arnold light and swap for a USD light then it won't work. It's not a particularly likely occurence perhaps, but it's so easy to switch between scenes in Gaffer that it doesn't seem impossible. I guess the same applies to the `if( object )` conditional at L1483.

If it's possible to always return a DelightLight and then to deal with the "didn't like the attributes" case in the DelightLight itself, that would be more robust.

Reply:
> If it's possible to always return a DelightLight

So that means we never return `nullptr`, right? And when switching between Arnold and USD, `DelightRenderer::light()` would not be hit again, but `DelightLight::attributes()` would, because the (non-existent) geometry never changed. Is that correct?

Are there any repercussions from not returning `nullptr` in the case of trying to render an Arnold light?

Reply:
> So that means we never return nullptr, right?

Yep.

> And when switching between Arnold and USD, DelightRenderer::light() would not be hit again, but DelightLight::attributes() would.

Yep.

> Are there any repercussions from not returning nullptr in the case of trying to render an Arnold light?

Not that I can think of. From the RenderController's perspective, it's just talking to a thing it can update attributes and transforms on, and it doesn't care if that's not attached to anything tangible yet.